File: Assignment/discount.py
price = int(input('Enter the price of the goods bought: '))
discount = 0  # Initialize discount variable

try:
    goods_number = int(input('Enter the number of goods bought: '))
     
    if goods_number >= 21:
        discount = float((8/100) * (price * goods_number))
    elif goods_number >= 11:
        discount = float((6/100) * (price * goods_number))
    elif goods_number >= 6:
        discount = float((4/100) * (price * goods_number))
    elif goods_number > 0:
        discount = float((2/100) * (price * goods_number))
    else:
        print("invalid goods number")

    gross_pay = float(price) * goods_number
    payable_amount = gross_pay - discount
    print(f'The amount to pay before discount is {gross_pay}\nThe discount for this price is {discount}\nThe price to pay after discount is {payable_amount}')

except ValueError:
    print('Invalid input for the number of goods bought. Please enter a valid integer.')

File: Tests/Discount.java
import java.util.Scanner;

public class Discount {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        
        System.out.println("Enter the number of goods bought: ");
        int goodsNumber = input.nextInt();
        
        // Validating Code 
        if (goodsNumber <= 0) {
            System.out.println("Your goods bought should be greater than 0!");
        } else {
            System.out.println("Enter the price of each good: ");
            float price = input.nextFloat();
            
            float grossPay = goodsNumber * price;
            float discount = 0;
            
            // Taking the lower limit in descending order
            if (goodsNumber >= 21) {
                discount = 0.08f * grossPay;
            } else if (goodsNumber >= 11) {
                discount = 0.06f * grossPay;
            } else if (goodsNumber >= 6) {
                discount = 0.04f * grossPay;
            } else {
                discount = 0.02f * grossPay;
            }
            
            float payableAmount = grossPay - discount;
            
            System.out.println("The amount payable for "+goodsNumber+ " goods bought before discount is: " + grossPay);
            System.out.println("The discount given is: "+ discount);
            System.out.println("The price to be paid after discount given is: " + payableAmount);
        }
        
        System.exit(0);
    }
}
